### DISCUSSION TITLE
"Key is stored in legacy trusted.gpg keyring" warning when using apt update

### DISCUSSION THREAD

**Discussion by yooakim** [ClickHouse/ClickHouse#40228] [General]
With Clickhouse installed on a Debian based system I get the following warning every time I use apt update:

```
> sudo apt update
W: https://packages.clickhouse.com/deb/dists/stable/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
```

Is there any specific reason why the key isn't stored in the recommended location?

What I did to resolve this is:

```
sudo apt-key export 2B48D754 | sudo gpg --dearmour -o /usr/share/keyrings/clickhouse.gpg
sudo apt-key del 2B48D754
```
Added `[signed-by=/usr/share/keyrings/clickhouse.gpg]` to `/etc/apt/sources.list.d/clickhouse.list`

```
sudo apt update
```

References:
* [What commands (exactly) should replace the deprecated apt-key?](https://askubuntu.com/questions/1286545/what-commands-exactly-should-replace-the-deprecated-apt-key)
* [apt-key deprecation warning when updating system](https://askubuntu.com/questions/1398344/apt-key-deprecation-warning-when-updating-system)

**Comment by DarkWanderer** [comment]
What would be the shortest path to get this set up from scratch? What steps should I execute instead of below?
```
sudo apt-get install -y apt-transport-https ca-certificates dirmgr
sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 8919F6BD2B48D754
echo "deb https://packages.clickhouse.com/deb stable main" | sudo tee /etc/apt/sources.list.d/clickhouse.list
```

(this can definitely be added to documentation)
